helpdesk_mgmt/models/helpdesk_ticket_team.py

```diff
@@ -36,40 +36,54 @@ class HelpdeskTeam(models.Model):
     ticket_ids = fields.One2many(
         comodel_name="helpdesk.ticket", inverse_name="team_id", string="Tickets",
     )
-    todo_ticket_ids = fields.One2many(
-        related="ticket_ids",
-        domain="[('closed', '=', False)]",
-        string="Todo tickets",
-        readonly=True,
-    )
     todo_ticket_count = fields.Integer(
-        string="Number of tickets", compute="_compute_todo_tickets"
+        string="Number of tickets", compute="_compute_todo_tickets", compute_sudo=True
     )
     todo_ticket_count_unassigned = fields.Integer(
-        string="Number of tickets unassigned", compute="_compute_todo_tickets"
+        string="Number of tickets unassigned",
+        compute="_compute_todo_tickets",
+        compute_sudo=True,
     )
     todo_ticket_count_unattended = fields.Integer(
-        string="Number of tickets unattended", compute="_compute_todo_tickets"
+        string="Number of tickets unattended",
+        compute="_compute_todo_tickets",
+        compute_sudo=True,
     )
     todo_ticket_count_high_priority = fields.Integer(
-        string="Number of tickets in high priority", compute="_compute_todo_tickets"
+        string="Number of tickets in high priority",
+        compute="_compute_todo_tickets",
+        compute_sudo=True,
```

Comment:
Why compute_sudo? Surely if a user is set to say "Own Tickets" the counts should reflect that. If sudo is required (which I doubt it is now with the removal of todo_ticket_ids as all the fields are non-stored) then do it at the write point, not at the read point.

It will be confusing for users to see 500 high priority tickets to open the list and just see 2.

Reply:
Those `compute_sudo=True` were added trying to solve #220 . If removed, I think the issue would remain unsolved. I didn't notice the case you've found. I'll take a look, thanks!

Reply:
@dalonsod I think you misdiagnosed the issue. For me the real issue was trying to write the non functional, related field todo_ticket_ids inside the function. But in any case, compute_sudo is definitely the "wrong" fix. Projects, CRM, Journals etc all successfully compute non stored functional fields for all users, tailored to their rights, without a need to elevate access rights.

Reply:
Comments attended, thanks @gdgellatly . Could you check if your performance issue is solved? The User role dashboard access issue seems to be solved too.

Reply:
@gdgellatly could you review this PR with the requested changes? Thanks!